File: types/property.ts
export type PropertyType = 'Hostel/PG' | 'Apartment';
export type ShareType = 'single' | 'double' | 'triple';

export interface Bed {
  id: string;
  occupied: boolean;
}

export interface Room {
  id: string;
  roomNumber: string;
  shareType: ShareType;
  beds: Bed[];
}

export interface Floor {
  id: string;
  label: string;
  rooms: Room[];
}

export interface Building {
  id: string;
  name: string;
  floors: Floor[];
}

export interface PropertyDetails {
  name: string;
  type: PropertyType | null;
  city: string;
}

export interface Property {
  id: string;
  name: string;
  type: PropertyType;
  city: string;
  buildings: Building[];
  totalRooms: number;
  totalBeds: number;
  createdAt: string;
}

export interface WizardState {
  currentStep: number;
  propertyDetails: PropertyDetails;
  buildings: Building[];
  allowedBedCounts: number[];
  editingPropertyId?: string | null;
}

File: api/app/schemas/property.py
from pydantic import BaseModel, Field
from typing import List, Optional, Literal

PropertyType = Literal['Hostel/PG', 'Apartments']
ShareType = Literal['single', 'double', 'triple']
BillingPeriod = Literal['monthly', 'weekly', 'hourly', 'yearly']

class BedPricing(BaseModel):
    bedCount: int
    dailyPrice: float
    monthlyPrice: float

class Bed(BaseModel):
    id: str
    occupied: bool

class Room(BaseModel):
    id: str
    roomNumber: str
    shareType: ShareType
    bedCount: Optional[int] = None
    beds: List[Bed]

class Floor(BaseModel):
    id: str
    label: str
    rooms: List[Room]

class Building(BaseModel):
    id: str
    name: str
    floors: List[Floor]

class PropertyDetails(BaseModel):
    name: str
    type: Optional[PropertyType]
    city: str
    area: Optional[str] = None

class Property(BaseModel):
    id: str
    name: str
    type: PropertyType
    city: str
    area: Optional[str] = None
    buildings: List[Building]
    bedPricing: List[BedPricing]
    totalRooms: int
    totalBeds: int
    createdAt: str
